src/scales/scale.category.js

```diff
@@ -25,7 +25,7 @@ const validIndex = (index, max) => index === null ? null : _limitValue(Math.roun
 function _getLabelForValue(value) {
   const labels = this.getLabels();
 
-  if (value >= 0 && value < labels.length) {
+  if (typeof value === 'number' && value >= 0 && value < labels.length) {
```

Comment:
The function should be used to use the internal values of the category scale and convert them to the string values.
The internal values are numbers. This feels like a wrong implementation of this function.

The type definition also specifies the input as a number and not a string
https://www.chartjs.org/docs/latest/api/classes/Scale.html#getlabelforvalue

I would find it more logical that there would be a guard against number inputs but that would possible be too breaking since inputting a number as a string will be seen as a number.

So the current implementation is the most correct in my eyes and this is not a bug.

Reply:
@LeeLenaleee but then the issue is here:
https://github.com/chartjs/Chart.js/blob/6372280085625b43ef34d7b70f3e86b063d22a10/src/core/core.datasetController.js#L694
```
  getLabelAndValue(index) {
    const meta = this._cachedMeta;
    const iScale = meta.iScale;
    const vScale = meta.vScale;
    const parsed = this.getParsed(index);
    return {
      label: iScale ? '' + iScale.getLabelForValue(parsed[iScale.axis]) : '',
      value: vScale ? '' + vScale.getLabelForValue(parsed[vScale.axis]) : ''
    };
  }
```
?

Reply:
Will add one. Currently I see this in my vscode extension but a small example doesn't show the issue. Will keep on analyzing.